internal/finder/versioner.go

```diff
@@ -101,14 +101,20 @@ func (v *Versioner) mostRecentKubectlVersionAvailableOrLatestFromUpstream() (sem
 // EnsureCompatibleKubectlAvailable ensures the kubectl binary with the specified
 // version is available on the system. It will return the full path to the
 // binary.
-func (v *Versioner) EnsureCompatibleKubectlAvailable(version semver.Version, allowDownload bool) (string, error) {
+func (v *Versioner) EnsureCompatibleKubectlAvailable(version semver.Version, allowDownload bool, useLatestIfNoCompatible bool) (string, error) {
 	bins := v.kFinder.AllKubectlBinaries(true)
 	kubectl, err := findCompatibleKubectl(version, bins)
 	if err == nil {
 		return kubectl.Path, nil
 	}
 
 	if !allowDownload {
+		if useLatestIfNoCompatible {
+			all := v.kFinder.AllKubectlBinaries(true /* reverseSort */)
```

Comment:
nit, just to stay consistent with the comment you wrote below (which I prefer):

```suggestion
			all := v.kFinder.AllKubectlBinaries(true) // newest-first
```